README.md

```diff
@@ -28,54 +28,47 @@ Require the library
 require 'graphql/batch'
 ```
 
-Define a GraphQL::Batch::Query derived class. Use group_key to specify which queries can be reduced into a batch query, and an execute class method which makes that batch query and passes the result to each individual query.
+Define a custom loader, which is initialized with arguments that are used for grouping and an execute method for doing the batch query. The load method can be overriden to coerce the type of the key.
 
 ```ruby
-class FindQuery < GraphQL::Batch::Query
-  attr_reader :model, :id
+class RecordLoader < GraphQL::Batch::Loader
+  attr_reader :model
 
-  def initialize(model, id, &block)
+  def initialize(model)
+    super()
     @model = model
-    @id = id
-    super(&block)
   end
 
-  # super returns the class name
-  def group_key
-    "#{super}:#{model.name}"
+  def load(key)
+    super(Integer(key))
   end
 
-  def self.execute(queries)
-    model = queries.first.model
-    ids = queries.map(&:id)
-    records_by_id = model.where(id: ids).index_by(&:id)
-    queries.each do |query|
-      query.complete(records_by_id[query.id])
-    end
+  def execute
+    model.find(keys).each { |record| fulfill(record.id, record) }
```

Comment:
is `keys` just a method that's available to you as a subclass? we should either document "use `keys` to access the batch of arguments" or just pass it into this method as a param